### ISSUE TITLE
Documentation detailing difference to/migration from Bitcoin Core missing

### ISSUE THREAD

**Issue by arnuschky** [conformal/btcd#208]
Unless I am mistaken, there's no document that clearly details the difference to Bitcoin Core, or describes a migration path. This would be really useful for people that consider moving away from Bitcoin Core.

Some questions this document might cover:
 - incompatibilities
 - differing default features and how to enable them
 - are there migration tools for Bitcoin Core's databases and wallets

I propose to collect related questions here to make writing this document easier.

**Comment by arnuschky** [comment]
Question (the one that actually drove me to look for said document): We're currently using bitcoind with txindex=1. However, I don't have to explicitly enable the index in btcd, right? I couldn't find any location where this is said explicitly.

**Comment by davecgh** [comment]
I don't believe there is a document, but that is a good idea.  Given the various uses, providing a specific migration guide might prove challenging since that would largely depend on how it's being used, but documenting any major differences seems like a reasonable thing to do.

For now, I'll answer your questions here and perhaps it can become the start of such a document.

The biggest difference is the fact that btcd **only**  provides chain services.  All wallet functionality has been split into the separate btcwallet daemon.  Bitcoin Core, on the other hand, currently combines everything into the daemon including both chain and wallet functionality (though I believe they have been working towards splitting the functionality as well).  Bitcoin-qt also combines in wallet GUI functionality.

As far as chain services:
* Incompatibilies:
  * The are no known incompatibilities in any of the core protocol or RPC calls.
  * However, btcd currently does not support JSON-RPC 2.0 batching mode which some things such as electrum use.  btcd will be adding support for this in the near future.
* Differing default features:
  * Bitcoin Core requires the `-txindex=1` option to enable the transaction index.  Btcd provides the transaction index by default and has no option to disable it.
  * Bitcoin Core does not have TLS protection enabled for RPC connections by default nor does it require it for remote connections.  Btcd has TLS enabled by default and **requires** it for remote connections (non-localhost).  TLS can be disable with btcd by using the `--notls` option, but only when listening solely on localhost interface.

* Are there migration tools for bitcoind's databases and wallets?
  * There are current no migration tools for converting the Bitcoin Core database directly to btcd.  However, a tool is provided to use bootstrap.dat (https://github.com/conformal/btcd/wiki#BootstrapDat)  It would be possible to use the Bitcoin Core linearize.py script to create a new bootstrap.dat file and then use the btcd addblock utility to import it.  That said, it's probably not worth the effort as it would most likely be faster to simply redownload the chain.
  * As mentioned above, btcd itself intentionally split the wallet functionality out of the core chain service daemon, so it does not have any wallet functionality, thus the wallet question does not apply directly.  However, [btcwallet](https://github.com/conformal/btcwallet) does support the WIF format that is dumped from the Bitcoin Core wallet via the `dumpprivkey` RPC call.  It can be imported with the `importprivkey` RPC call that Bitcoin Core uses.
  * The differences in the Bitcoin Core wallet functionality and btcwallet really will probably end up needed its own document as btcwallet is different than Bitcoin Core's wallet in many ways, and it currently has open pull requests that will be bring a ton of new functionality to it such as BIP0032 hierarchical deterministic keys and proper accounts support which will be further differences.

Also, I'd like to point out that btcwallet is a daemon just like btcd.  In other words it's intended for programmatic access.  That means it does not have a shiny GUI like Bitcoin-qt/Armory/etc that most end users are used to interfacing with.  There is the [btcgui](https://github.com/conformal/btcgui) project which is a *proof of concept* on how such a GUI can be built on top of btcwallet, however, since btcgui is only a proof of concept, it is no where near ready for end users.

**Comment by arnuschky** [comment]
Wow, amazingly comprehensive answer (in such a short time). Thank you, that answers my question relating `-txindex`. Good to know for tls rpc, I can see this being a major trap when moving an existing (and insecure ;)) installation from Bitcoin Core to btcd.

**Comment by davecgh** [comment]
This was not supposed to be closed.

**Comment by manan19** [comment]
For a temporary solution,

>set port=8339 in bitcoin.conf and start bitcoind
>btcd --testnet --addpeer=127.0.0.1:8339 --maxpeers=1
